cpu/esp_common/exceptions.c

```diff
@@ -80,8 +85,42 @@ static const char* exception_names [] =
     "Coprocessor7Disabled",        /* 39 */
 };
 
+#ifdef MCU_ESP8266
+
+/* Enables the IROM cache if needed. */
+static void IRAM ensure_flash_cache_enabled(void)
+{
+    if (!(READ_PERI_REG(CACHE_FLASH_CTRL_REG) & CACHE_READ_EN_BIT)) {
```

Comment:
Why don't we just use the existing `Cache_Read_Enable_New` function, regardless of whether the cache is disabled or not? This would avoid low-level register access outside the vendor code.

Reply:
I don't think we can do Wait_SPI_Idle() if the cache is enabled. This exception can run at pretty much any time, so this seemed more robust. I can replace the two other calls with a single call to Cache_Read_Enable_New() but I thought it was only included if the flash module was included (`esp_idf_spi_flash`), I see that for esp8266 it is unconditionally included so I can change it.

Reply:
> I can replace the two other calls with a single call to Cache_Read_Enable_New()

@iosabi Will you provide these changes so that we can merge it?

Reply:
@iosabi, ping!